### DISCUSSION TITLE
How to pass AWS Session as environment dependency in TileFactory ?

### DISCUSSION THREAD

**Discussion by AbhishekD15** [developmentseed/titiler#999] [Q&A]
I have a private s3 bucket. I would like to access the Object URL from the routes defined in Tilefactory.
I have a role that I am using to create an AWS session. I would like TilerFactory to include AWS session in the environment. What I am doing atm is creating a custom TileFactory and using the  environment_dependency top inject the AWS session.

from rasterio.session import AWSSession
```bash
class CustomTilerFactory(TilerFactory):
    
    def environment_dependency(self):
        session = get_assumed_role_session()
        aws_session = AWSSession(session)
        return {'session': aws_session}
```

The `get_assumed_role_session()` returns a session using the role to access the private bucket.

Unfortunately still getting 403. Could someone please guide me ? Thanks

**Comment by vincentsarago** [comment]
I don't think this will change the result but usually I would do it like:

```python
def environment_dependency():
    session = get_assumed_role_session()
    aws_session = AWSSession(session)
    return {'session': aws_session}


endpoints = TilerFactory(environment_dependency=environment_dependency)
```

**Reply by AbhishekD15** [reply]
Nope, I am still getting `rasterio.errors.RasterioIOError: HTTP response code: 403`. I am just wondering once we create an instance of `Tilerfactory` , is there a property or a way to check the env dependencies. I just want to verify what I am injecting is what is used in the routes.
